### PR TITLE
Created initial dialog box expansion for individual posts

### PR DESCRIPTION
**Pull request by Ilithor**

### CODE REVIEW

File: server/handlers/find.js [hunk 1]
```diff
@@ -120,19 +120,13 @@ export const findNotificationByRecipient = async recipient => {
  * @return {Promise<UserComment> | NotificationNotFound}
  */
 export const findCommentByPostId = async _id => {
-  let comment = [];
-  let error = {};
-  comment = await Comment.find({
+  let commentList = [];
+  commentList = await Comment.find({
```
Comment (jeffersoneagley) [inline server/handlers/find.js]: you could just directly return Comment.find({
blablabla
}).blabla;

Comment (jeffersoneagley) [reply server/handlers/find.js]: because this just waits for a promise to return

File: server/handlers/post.js [hunk 2]
```diff
@@ -38,20 +38,16 @@ export const getPostList = (req, res, next) => {
...
+    const postData = { ...returnPost._doc };
+    postData.commentList = [];
     findCommentByPostId(returnPost._id)
       .then(post => {
-        if (post.comment) {
-          return res.status(404).json({ error: post.comment });
-        }
         post.forEach(doc => {
-          postData.comment.push(doc);
+          postData.commentList.push(doc);
```
Comment (jeffersoneagley) [inline server/handlers/post.js]: I'm confused what we're mapping here... is this just converting an object to an array? maybe add some types here so it's clear what's happening. I think theres stuff here that could be converted to a 1-liner from lodash like
postData.commentList = values(post)

File: src/redux/types.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ export const LOADING_USER = 'LOADING_USER';
 // UI reducer types
 export const SET_ERRORS = 'SET_ERRORS';
 export const LOADING_UI = 'LOADING_UI';
+export const STOP_LOADING_UI = 'STOP_LOADING_UI';
```
Comment (jeffersoneagley) [inline src/redux/types.js]: IS_UI_LOADING

File: src/redux/reducers/uiReducer.js [hunk 4]
```diff
@@ -24,6 +29,11 @@ export default function(state = initialState, action) {
         ...state,
         isLoading: true,
       };
+    case STOP_LOADING_UI:
+      return {
+        ...state,
+        isLoading: false,
```
Comment (jeffersoneagley) [inline src/redux/reducers/uiReducer.js]: IS_UI_LOADING:
return {
...state,
isLoading: action.payload
}

File: server/handlers/post.js (hunk 2 above)
Comment (Ilithor) [reply server/handlers/post.js]: The data being returned from the findCommentByPostId is an array. Taking that into account, I might be able to just spread it's contents into postData.commentList.

File: server/handlers/find.js (hunk 1 above)
Comment (Ilithor) [reply server/handlers/find.js]: fixed

File: server/handlers/post.js (hunk 2 above)
Comment (Ilithor) [reply server/handlers/post.js]: fixed

File: src/redux/types.js (hunk 3 above)
Comment (Ilithor) [reply src/redux/types.js]: fixed

File: src/redux/reducers/uiReducer.js (hunk 4 above)
Comment (Ilithor) [reply src/redux/reducers/uiReducer.js]: fixed

File: src/components/post/postList/postDialog/PostDialog.js [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => {
+    getPost(postId);
+    setOpen(true);
+  };
+  const handleClose = () => setOpen(false);
+  const makeDialogContentEditor = () => {
+    if (UI.isLoading) {
+      return <CircularProgress size={120} />;
+    } else {
```
Comment (jeffersoneagley) [inline src/components/post/postList/postDialog/PostDialog.js]: unneeded else

File: src/components/post/postList/postDialog/PostDialog.js [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+  postId,
+  userHandle,
+  UI = {},
+  post = {},
+  getPost,
+}) => {
+  const [open, setOpen] = useState(false);
+  const handleOpen = () => {
+    getPost(postId);
+    setOpen(true);
```
Comment (jeffersoneagley) [inline src/components/post/postList/postDialog/PostDialog.js]: Nice use of lazy loading strategy 🔃

File: src/components/post/postList/postDialog/PostDialog.js (hunk 5 above)
Comment (Ilithor) [reply src/components/post/postList/postDialog/PostDialog.js]: fixed

**Pull request merged**